tests/conftest.py

```diff
@@ -79,6 +184,38 @@ def make(*args, **kwargs):
         app_.cleanup()
 
 
+class SharedResult(object):
+    cache = {}
+
+    def store(self, key, app_):
+        if key in self.cache:
+            return
```

Comment:
Because the warning (and status) output of the second build differs from that of the first one in most cases, so saving second one will change the test result.